src/GafferSceneUI/SetMembershipInspector.cpp

```diff
@@ -216,14 +216,40 @@ IECore::ConstObjectPtr SetMembershipInspector::value( const GafferScene::SceneAl
 	return matchResult & IECore::PathMatcher::Result::ExactMatch ? new BoolData( true ) : nullptr;
 }
 
-IECore::ConstObjectPtr SetMembershipInspector::fallbackValue( const GafferScene::SceneAlgo::History *history ) const
+IECore::ConstObjectPtr SetMembershipInspector::fallbackValue( const GafferScene::SceneAlgo::History *history, std::string &description ) const
 {
 	const auto &path = history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
 	ConstPathMatcherDataPtr setMembers = history->scene->set( m_setName );
 
 	auto matchResult = (PathMatcher::Result)setMembers->readable().match( path );
 
-	return new BoolData( matchResult & IECore::PathMatcher::Result::AncestorMatch );
+	const bool ancestorMatch = matchResult & IECore::PathMatcher::Result::AncestorMatch;
+	if( ancestorMatch )
+	{
+		ScenePlug::ScenePath currentPath( history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName ) );
+		if( currentPath.size() > 1 )
+		{
+			// We start the inheritance search from the parent in order to return the value that
+			// would be inherited if the original location wasn't a member of the inspected set.
+			currentPath.pop_back();
+		}
+
+		while( !currentPath.empty() )
```

Comment:
We do currently allow `/` to be a set member, so I think we need to check even the empty case. I think using do-while would mean we can even remove the initial `pop_back()`?

```
do
{
    currentPath.pop_back();
    ...
} while( !currentPath.empty() )
```

Reply:
Switched to `do ... while` and squashed to d1c5ac3. 

Also expanded `SetMembershipInspectorTest.testFallbackValue` to account for `/` being allowed as a set member.